addons/account/models/account_move_line.py

```diff
@@ -1108,7 +1108,7 @@ def _compute_analytic_distribution(self):
     def _inverse_partner_id(self):
         self._conditional_add_to_compute('account_id', lambda line: (
             line.display_type == 'payment_term'  # recompute based on settings
-            or (line.move_id.is_invoice(True) and line.display_type == 'product' and not line.product_id)  # recompute based on most used account
+            or (line.move_id.is_invoice(True) and line.display_type == 'product' and not line.product_id and not line.account_id)  # recompute based on most used account
```

Comment:
Actually, this fix https://github.com/odoo/odoo/commit/55826a514b0a25aff03dde2d4bc9f520fa295203 should have been removing this line

It doesn't serve any purpose if we don't want to recompute on misc entries

Reply:
Thanks for the review.

I've removed the line.